cclib/parser/gaussianparser.py

```diff
@@ -1909,47 +1910,109 @@ def natural_orbital_single_spin_parsing(inputfile, updateprogress_title):
+            has_spin = 'spin' in line.lower()
+            has_charges = 'charges' in line.lower()
             if has_charges and not hasattr(self, "atomcharges"):
                 self.atomcharges = {}
-
             if has_spin and not hasattr(self, "atomspins"):
                 self.atomspins = {}
-
             ones = next(inputfile)
-
             charges = []
             spins = []
-            nline = next(inputfile)
-            while not "Sum of" in nline:
-                if has_charges:
-                    charges.append(float(nline.split()[2]))
-
-                if has_spin and has_charges:
-                    spins.append(float(nline.split()[3]))
-
-                if has_spin and not has_charges:
-                    spins.append(float(nline.split()[2]))
-
-                nline = next(inputfile)
-
-            if "Mulliken" in line:
+            is_sum = 'summed' in line
+            # Iterate over each line and append values to a list 
+            # based on whether they are charges or spins. 
+            if is_sum:
+                for i in self.atomnos:
+                    # currently bug exists where files with translation vectors report
+                    # an extra atom with atomnumber -2 in self.atomnos, so must ignore 
+                    # this by passing whenever i in self.atomnos == -2.
+                    if i == -2:
+                        pass
+                    # For lists of summed charges or spins, a value
+                    # of 0 is added if the atom is a hydrogen.
+                    elif i == 1:
+                        if has_charges:
+                            charges.append(float(0))
+                        if has_spin and has_charges:
+                            spins.append(float(0))
+                        if has_spin and not has_charges:
+                            spins.append(float(0))
+                    else:
+                        nline = next(inputfile)
+                        # Some older versions of Gaussian already include 
+                        # hydrogens with value 0 for summed charges or 
+                        # spins, so these should be ignored.
+                        while nline.split()[1] == "H":
+                            nline = next(inputfile)
+                        if has_charges:
+                            charges.append(float(nline.split()[2]))
+                        if has_spin and has_charges:
+                            spins.append(float(nline.split()[3]))
+                        if has_spin and not has_charges:
+                            spins.append(float(nline.split()[2]))
+            else:
+                for i in self.atomnos:
+                    # Ignore translation vectors.
+                    if i == -2:
+                        pass
+                    else:
+                        nline = next(inputfile)
+                        if has_charges:
+                            charges.append(float(nline.split()[2]))
```

Comment:
Can you move the split out of the conditional, so there aren't three separate split calls?

Reply:
Yeah, I've done this for both.